ta/remoteproc/remoteproc_core.c

```diff
@@ -557,7 +559,12 @@ static paddr_t remoteproc_da_to_pa(uint32_t da, size_t size,
 		return 0;
 	}
 
-	return (paddr_t)reg_pair_to_64(params[3].value.b, params[3].value.a);
+	pa = (paddr_t)reg_pair_to_64(params[3].value.b, params[3].value.a);
+
+	/* On success, assert that the pa address is not 0 */
+	assert(res != TEE_SUCCESS || pa != 0);
```

Comment:
You can remove the test on `res`. It's always `TEE_SUCCESS` when we're here.

Reply:
done